filament/backend/src/metal/MetalDriver.mm

```diff
@@ -328,13 +328,17 @@
     ASSERT_POSTCONDITION(!depth.handle || any(targetBufferFlags & TargetBufferFlags::DEPTH),
             "The DEPTH flag was specified, but no depth texture provided.");
 
-    construct_handle<MetalRenderTarget>(rth, mContext, width, height, samples,
-            colorAttachments, depthAttachment);
+    MetalRenderTarget::Attachment stencilAttachment = {};
+    if (stencil.handle) {
```

Comment:
I think it's not enough to test `stencil.handle` here, you also need to test `targetBufferFlags`.

Reply:
Is it valid for `stencil.handle` to be a valid handle but `targetBufferFlags` not have the stencil bit set?

Reply:
I would say yes. It's really `targetBufferFlags` that tells you if you should use it or not.